src/screens/KeyBasedAssetTransfer/KeyBasedAssetTransferEditAmount.js

```diff
@@ -0,0 +1,252 @@
+import type { NavigationScreenProp } from 'react-navigation';
+
+// actions
+import {
+  addKeyBasedAssetToTransferAction,
+  fetchAvailableBalancesToTransferAction,
+  removeKeyBasedAssetToTransferAction,
+} from 'actions/keyBasedAssetTransferActions';
+
+// constants
+import { COLLECTIBLES } from 'constants/assetsConstants';
+
+// components
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import { Wrapper } from 'components/Layout';
+import SearchBlock from 'components/SearchBlock';
+import ListItemWithImage from 'components/ListItem/ListItemWithImage';
+import EmptyStateParagraph from 'components/EmptyState/EmptyStateParagraph';
+import { BaseText } from 'components/Typography';
+import TextInput from 'components/Input';
+
+// utils
+import { fontSizes, spacing, fontStyles } from 'utils/variables';
+import { parseNumber, isValidNumber, formatFullAmount, isValidNumberDecimals } from 'utils/common';
+import { getBalance } from 'utils/assets';
+import { themedColors } from 'utils/themes';
+
+// types
+import type { Balances, KeyBasedAssetTransfer, AssetData } from 'models/Asset';
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
+
+
+type Props = {
+  navigation: NavigationScreenProp<*>,
+  fetchAvailableBalancesToTransfer: () => void,
+  removeKeyBasedAssetToTransfer: (assetData: AssetData) => void,
+  addKeyBasedAssetToTransfer: (assetData: AssetData, amount: number) => void,
+  keyBasedAssetsToTransfer: KeyBasedAssetTransfer[],
+  availableBalances: Balances,
+  isFetchingAvailableBalances: boolean,
+};
+
+const ErrorHolder = styled.View`
+  width: 100%;
+  padding: 0 ${spacing.mediumLarge}px;
+`;
+
+const ErrorText = styled(BaseText)`
+  color: ${themedColors.negative};
+  ${fontStyles.small};
+  width: 100%;
+  text-align: right;
+`;
+
+const AmountInputWrapper = styled.View`
+  height: 70px;
+  justify-content: center;
+  min-width: 180px;
+`;
+
+const AmountInput = styled(TextInput)`
+  ${fontSizes.small};
+  text-align: right;
+  max-width: 200px;
+`;
+
+const KeyBasedAssetTransferEditAmount = ({
+  keyBasedAssetsToTransfer,
+  navigation,
+  isFetchingAvailableBalances,
+  availableBalances,
+  fetchAvailableBalancesToTransfer,
+  removeKeyBasedAssetToTransfer,
+  addKeyBasedAssetToTransfer,
+}: Props) => {
+  const [inSearchMode, setInSearchMode] = useState(false);
+  const [updatedValues, setUpdatedValues] = useState({});
+  const [errorMessages, setErrorMessages] = useState({});
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const onAvailableBalancesRefresh = () => {
+    if (isFetchingAvailableBalances) return;
+    fetchAvailableBalancesToTransfer();
+  };
+
+  const handleAmountChange = (text: string, assetData: AssetData) => {
+    const { token: symbol, decimals } = assetData;
+    const balance = getBalance(availableBalances, symbol);
+    const amountFormatted = text.toString().replace(/,/g, '.');
+    const amount = parseNumber(text);
```

Comment:
Will this `text` always be something that is parseable? Just thinking about the worse case scenario as the failure of this function isn't elegant 😂

Reply:
Yes, it's always a `string` based on `onChangeText` return for text input component.